include/podio/ROOTFrameReader.h

```diff
@@ -0,0 +1,97 @@
+#ifndef PODIO_ROOTFRAMEREADER_H
+#define PODIO_ROOTFRAMEREADER_H
+
+#include "podio/CollectionBranches.h"
+#include "podio/ROOTRawData.h"
+#include "podio/podioVersion.h"
+
+#include "TChain.h"
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+// forward declarations
+class TClass;
+// class TChain;
+class TFile;
+class TTree;
+
+namespace podio {
+
+class EventStore;
+class CollectionBase;
+class CollectionIDTable;
+class GenericParameters;
+struct CollectionReadBuffers;
+
+/**
+ * This class has the function to read available data from disk
+ * and to prepare collections and buffers.
+ **/
+class ROOTFrameReader {
+
+public:
+  ROOTFrameReader() = default;
+  ~ROOTFrameReader() = default;
+
+  // non-copyable
+  ROOTFrameReader(const ROOTFrameReader&) = delete;
+  ROOTFrameReader& operator=(const ROOTFrameReader&) = delete;
+
+  void openFile(const std::string& filename);
+
+  void openFiles(const std::vector<std::string>& filenames);
+
+  /// Read all collections requested
+  std::unique_ptr<podio::ROOTRawData> readNextFrame(const std::string& category);
```

Comment:
what behaviour do you want to have in case of end of iteration? The doc string should reflect that